zebra-state/src/lib.rs

```diff
@@ -79,12 +86,19 @@ impl Config {
             Network::Mainnet => "mainnet",
             Network::Testnet => "testnet",
         };
-        let path = self.cache_dir.join(net_dir).join("state");
 
-        sled::Config::default()
-            .path(path)
-            .cache_capacity(self.memory_cache_bytes)
-            .mode(sled::Mode::LowSpace)
+        if self.ephemeral {
+            sled::Config::default()
+                .cache_capacity(self.memory_cache_bytes)
+                .mode(sled::Mode::LowSpace)
+                .temporary(self.ephemeral)
+        } else {
+            let path = self.cache_dir.join(net_dir).join("state");
+            sled::Config::default()
+                .path(path)
+                .cache_capacity(self.memory_cache_bytes)
+                .mode(sled::Mode::LowSpace)
+        }
```

Comment:
Looks good, but let's avoid duplicating code here.

Otherwise the other options in the ephemeral and path configs could get out of sync, when we add or modify those options.

```suggestion
        let config = sled::Config::default()
                .cache_capacity(self.memory_cache_bytes)
                .mode(sled::Mode::LowSpace);
        if self.ephemeral {
                config.temporary(self.ephemeral)
        } else {
            let path = self.cache_dir.join(net_dir).join("state");
            config.path(path)
        }
```